### PR TITLE
Compatibility with new admin UI

### PR DESCRIPTION
**Pull request by andrasguseo**
Don't let the branch name fool you, this is version 1.2.2.

The extra frequencies for trashing old events wasn't showing up on the new admin UI. This should fix it.
Also, some functionality was reworked a bit to make it simpler or more compatible.

### CODE REVIEW

File: src/Tec/Plugin.php [hunk 1]
```diff
@@ -193,25 +186,20 @@ function reschedule_trash_or_del_event_cron( $cron ) {
...
+		elseif ( str_starts_with( $interval, 'hour' ) ) {
+			return 'hourly';
 		}
 
-		$scheduled = wp_next_scheduled( $cron );
-
-		if ( $scheduled && $frequency !== wp_get_schedule( $cron ) ) {
-			wp_unschedule_event( $scheduled, $cron );
-			wp_schedule_event( time(), $frequency, $cron );
-		}
```
Comment (dpanta94) [inline src/Tec/Plugin.php:215]: Why are we no longer scheduling events ?

File: src/Tec/Plugin.php [hunk 2]
```diff
@@ -144,43 +148,32 @@ function option_filter( $fields ) {
...
 	 */
-	function reschedule_trash_or_del_event_cron( $cron ) {
-		if ( 'tribe_trash_event_cron' == $cron ) {
-			$time = tribe_get_option( 'trash-past-events', 43200 );
-		} else {
-			$time = tribe_get_option( 'delete-past-events', 43200 );
-		}
+	function reschedule_crons() {
+		// Get the setting, default to 1 month.
+		$time = tribe_get_option( 'trash-past-events', 1 );
```
Comment (dpanta94) [inline src/Tec/Plugin.php:175]: We stopped controlling the delete-past-events frequency.

File: src/Tec/Plugin.php [hunk 3]
```diff
@@ -144,43 +148,32 @@ function option_filter( $fields ) {
...
-	 * @param string $cron The slug of the cron.
+	 * @since 1.2.2 Adjust cron frequency calculation.
+	 *              Remove $cron parameter.
 	 *
-	 * @return void
+	 * @return string The frequency string how often the cron should run.
 	 */
-	function reschedule_trash_or_del_event_cron( $cron ) {
-		if ( 'tribe_trash_event_cron' == $cron ) {
-			$time = tribe_get_option( 'trash-past-events', 43200 );
```
Comment (dpanta94) [inline src/Tec/Plugin.php:180]: The default was twicedaily now it became once a month. 43200 is half a day in seconds

File: src/Tec/Plugin.php [hunk 4]
```diff
@@ -144,43 +148,32 @@ function option_filter( $fields ) {
...
-			'2|year'    => esc_html__( '2 years', 'the-events-calendar' ),
-			'3|year'    => esc_html__( '3 years', 'the-events-calendar' ),
+			'1'         => esc_html__( '1 month', 'the-events-calendar' ),
+			'3'         => esc_html__( '3 months', 'the-events-calendar' ),
+			'6'         => esc_html__( '6 months', 'the-events-calendar' ),
+			'9'         => esc_html__( '9 months', 'the-events-calendar' ),
+			'12'        => esc_html__( '1 year', 'the-events-calendar' ),
+			'24'        => esc_html__( '2 years', 'the-events-calendar' ),
+			'36'        => esc_html__( '3 years', 'the-events-calendar' ),
 		];
```
Comment (dpanta94) [inline src/Tec/Plugin.php:158]: We have to make sure these are all registered in WP's cron_schedules so they can be identified from a string. e.g.

`tribe-every15mins` is being registered right [here](https://github.com/the-events-calendar/the-events-calendar/blob/feat/slr-support/src/Tribe/Aggregator/Cron.php#L205)

File: src/Tec/Plugin.php [hunk 5]
```diff
@@ -144,43 +148,32 @@ function option_filter( $fields ) {
...
-	 * @return void
+	 * @return string The frequency string how often the cron should run.
 	 */
-	function reschedule_trash_or_del_event_cron( $cron ) {
-		if ( 'tribe_trash_event_cron' == $cron ) {
-			$time = tribe_get_option( 'trash-past-events', 43200 );
-		} else {
-			$time = tribe_get_option( 'delete-past-events', 43200 );
-		}
+	function reschedule_crons() {
```
Comment (dpanta94) [inline src/Tec/Plugin.php:173]: The logic of this function should change to handle both hooks and return a dynamic string which would represent an already registered cron schedule. We would determine this string based on the user's selected option.

**Review by dpanta94 — CHANGES_REQUESTED** [review]

File: src/Tec/Plugin.php [hunk 6]
```diff
@@ -144,43 +148,32 @@ function option_filter( $fields ) {
 			'3|day'     => esc_html__( '3 days', 'the-events-calendar' ),
 			'1|week'    => esc_html__( '1 week', 'the-events-calendar' ),
 			'2|week'    => esc_html__( '2 weeks', 'the-events-calendar' ),
-			'1|month'   => esc_html__( '1 month', 'the-events-calendar' ),
-			'3|month'   => esc_html__( '3 months', 'the-events-calendar' ),
-			'6|month'   => esc_html__( '6 months', 'the-events-calendar' ),
-			'9|month'   => esc_html__( '9 months', 'the-events-calendar' ),
-			'1|year'    => esc_html__( '1 year', 'the-events-calendar' ),
-			'2|year'    => esc_html__( '2 years', 'the-events-calendar' ),
-			'3|year'    => esc_html__( '3 years', 'the-events-calendar' ),
```
Comment (dpanta94) [inline src/Tec/Plugin.php:153]: Not easy to change those. Think that these could already live inside user's DBs. I would suggest leaving them as they were.

The alternatives are either migration or having fallbacks in place.

File: src/Tec/Plugin.php (hunk 1 above)
Comment (andrasguseo) [reply src/Tec/Plugin.php:215]: The trash cron is set up automatically by TEC.
We have a filter that allows us to change the frequency of the trash cron, and that filter is being used in the extension.
(Line 96)

File: src/Tec/Plugin.php [hunk 7]
```diff
@@ -144,43 +148,32 @@ function option_filter( $fields ) {
...
+			'1'         => esc_html__( '1 month', 'the-events-calendar' ),
+			'3'         => esc_html__( '3 months', 'the-events-calendar' ),
+			'6'         => esc_html__( '6 months', 'the-events-calendar' ),
+			'9'         => esc_html__( '9 months', 'the-events-calendar' ),
+			'12'        => esc_html__( '1 year', 'the-events-calendar' ),
+			'24'        => esc_html__( '2 years', 'the-events-calendar' ),
+			'36'        => esc_html__( '3 years', 'the-events-calendar' ),
 		];
 		$fields['delete-past-events']['options'] = $new_values;
 		$fields['trash-past-events']['options']  = $new_values;
```
Comment (dpanta94) [inline src/Tec/Plugin.php]: Lets delete what we don't interfere with.

File: src/Tec/Plugin.php (hunk 2 above)
Comment (dpanta94) [inline src/Tec/Plugin.php:175]: The default should remain to half a day for backwards compatibility. Here you are defaulting to a month, affecting customers that already use the extension without a value present in the db.

**Review by dpanta94 — CHANGES_REQUESTED** [review]

File: src/Tec/Plugin.php (hunk 2 above)
Comment (andrasguseo) [reply src/Tec/Plugin.php:175]: That `1 (month)` is the value for how old events should be deleted. It's not the value for the cron frequency.
Based on this `1 (month)` the cron schedule will be set to `twicedaily` on line 203. Basically, this sets the default value for the cron frequency.

File: src/Tec/Plugin.php (hunk 7 above)
Comment (andrasguseo) [reply src/Tec/Plugin.php]: Removed.

**Review by dpanta94 — APPROVED** [review]

**Pull request merged**